addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/handler/YamahaZoneThingHandler.java

```diff
@@ -0,0 +1,606 @@
+                        return;
+                    }
+                    if (((UpDownType) command) == UpDownType.UP) {
+                        inputWithNavigationControl.goUp();
+                    } else {
+                        inputWithNavigationControl.goDown();
+                    }
+                    break;
+
+                case YamahaReceiverBindingConstants.CHANNEL_NAVIGATION_LEFTRIGHT:
+                    if (inputWithNavigationControl == null) {
+                        logger.error("Channel " + id + " not working with this input!");
+                        return;
+                    }
+                    if (((UpDownType) command) == UpDownType.UP) {
+                        inputWithNavigationControl.goLeft();
+                    } else {
+                        inputWithNavigationControl.goRight();
+                    }
+                    break;
+
+                case YamahaReceiverBindingConstants.CHANNEL_NAVIGATION_SELECT:
+                    if (inputWithNavigationControl == null) {
+                        logger.error("Channel " + id + " not working with this input!");
+                        return;
+                    }
+                    inputWithNavigationControl.selectCurrentItem();
+                    break;
+                case YamahaReceiverBindingConstants.CHANNEL_NAVIGATION_BACK:
+                    if (inputWithNavigationControl == null) {
+                        logger.error("Channel " + id + " not working with this input!");
+                        return;
+                    }
+                    inputWithNavigationControl.goBack();
+                    break;
+
+                case YamahaReceiverBindingConstants.CHANNEL_NAVIGATION_BACKTOROOT:
+                    if (inputWithNavigationControl == null) {
+                        logger.error("Channel " + id + " not working with this input!");
+                        return;
+                    }
+                    inputWithNavigationControl.goToRoot();
+                    break;
+
+                case YamahaReceiverBindingConstants.CHANNEL_PLAYBACK_PRESET:
+                    if (inputWithPlayControl == null) {
+                        logger.error("Channel " + id + " not working with this input!");
+                        return;
+                    }
+
+                    inputWithPlayControl.selectItemByPresetNumber(((DecimalType) command).intValue());
+                    break;
+
+                case YamahaReceiverBindingConstants.CHANNEL_PLAYBACK:
+                    if (inputWithPlayControl == null) {
+                        logger.error("Channel " + id + " not working with this input!");
+                        return;
+                    }
+
+                    if (command instanceof PlayPauseType) {
+                        PlayPauseType t = ((PlayPauseType) command);
+                        switch (t) {
+                            case PAUSE:
```

Comment:
Two nestings are fine with me in this case, and is somehow enforced by OH on me anyway. First I have to check for the channel ID and then for the command, which if it is an enum, is best processed by a switch. Personally I don't like to split the handleCommand() functionality.